src/app-layer-ssl.c

```diff
@@ -143,11 +143,93 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            input += 2;
+
+            /* skip random */
+            input += 32;
+
+            /* skip session id */
+            uint8_t session_id_length = *(input++);
+            input += session_id_length;
```

Comment:
Its possible to get here and dereference input with much less than 34 bytes. Check the bounds of input as you progress, and perhaps create a #define for random length, and version length.